pmd-java/src/main/resources/category/java/bestpractices.xml

```diff
@@ -2171,21 +2171,27 @@ a block `{}` is sufficient.
         <properties>
             <property name="xpath">
                 <value>
-//DoStatement[Expression/PrimaryExpression/PrimaryPrefix/Literal/BooleanLiteral] |
+<![CDATA[
+//DoStatement[Expression/(.|*)/PrimaryExpression/PrimaryPrefix/Literal/BooleanLiteral] |
```

Comment:
The path `(.|*)` will match 0 or exactly one level. Maybe we can write it as `Expression//PrimaryExpression...`, which will match 0 or more levels. That however might create more false positives, not sure.

Could you also check, what effect parentheses have? E.g. `do {} while ( (false|false) )`